pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -628,9 +635,70 @@ func (r *Reconciler) computeDataHashAnnotation(existing, secret *v1.Secret) stri
 func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, opts controller.Options) error {
 	r.recorder = mgr.GetEventRecorderFor("external-secrets")
 
+	// Index .Spec.Target.Name to reconcile ExternalSecrets effectively when secrets have changed
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &esv1beta1.ExternalSecret{}, externalSecretSecretNameKey, func(obj client.Object) []string {
+		es := obj.(*esv1beta1.ExternalSecret)
+
+		if name := es.Spec.Target.Name; name != "" {
+			return []string{name}
+		}
+		return []string{es.Name}
+	}); err != nil {
+		return err
+	}
+
```

Comment:
I am not at all familiar with how `IndexField` behaves - but what happens here if `spec.target.name` is defined afterwards?

Reply:
The indexer is used from the informer for `esv1beta1.ExternalSecret`, so whenever `esv1beta1.ExternalSecret` changes, the informer watches the change and updates the index if necessary 🙂